src/sql/workbench/services/notebook/browser/models/clientSession.ts

```diff
@@ -332,7 +332,18 @@ export class ClientSession implements IClientSession {
 	 * this is a no-op, and resolves with `false`.
 	 */
 	restart(): Promise<boolean> {
-		throw new Error('Not implemented');
+		if (!this._session?.kernel) {
```

Comment:
This doesn't seem to match the doc comments about starting a kernel with the last run kernel name